### DISCUSSION TITLE
Welcome to Discussions!

### DISCUSSION THREAD

**Discussion by postspectacular** [thi-ng/umbrella#374] [General]
## 👋 Welcome!

_tl;dr_ I warmly invite everyone interested to use this place to connect with other members of this small community:

- Ask questions you’re wondering about or things you need help with
- Share your ideas or interesting things related to the project(s)
- Engage with other community members, provide help, help answer questions
- Welcome others, be open-minded. Remember this is a community we build together!

To get started, comment below with an introduction of yourself and tell us about what you do with this community.

### Background

In 2018 I created a [Discord server](https://discord.gg/JhYcmBw) for this project. Slowly it grew into a great place to meet & hang out with other people to discuss various topics, issues & inspiring research related to the (by now >175) projects in this monorepo, to provide help, share links, create examples... Alas, due to workload, stressful international relocation (potent mix of Brexit & COVID), giving up my day job, working on other new projects, and - last but not least - misplacing my 2FA keys, I completely abandoned Discord sometime in 2021...

Meanwhile a lot of things happened. Some of these driven by external forces, life, but also a general desire to return more to the open web and reduce reliance on increasingly abusive platforms. In late 2022 I stopped using Twitter, set up a [Mastodon](https://mastodon.thi.ng/) instance ([also for this project](https://mastodon.thi.ng/tags/thingumbrella)) and started [migrating/consolidating my blog](https://github.com/thi-ng/blog) away from [Medium](https://medium.com/@thi.ng). Even though I managed to regain access to Discord recently, I've always been dubious about its closed off, invisible/unsearchable (from the web), walled garden nature and hoped it would evolve to shed some of its informational black hole characteristics (how naive!).

By now I'm convinced Discord absolutely isn't the right choice to invest any more time in to help growing & evolving a communal base camp & effective socially-led knowledge base. I truly think everyone will be better served by a more oldskool forum-based format, having open discussions directly on Github, alongside with the rest of the project. I realize this is somewhat hypocritical, since GH itself isn't exactly a "free" platform, but at the very least, it offers much better integration with the code base & will make migration much easier than from Discord, should the need arise in the future...

I want to stress that I believe in distributed, self-organizing open source communities and as such people are of course free to continue using the existing Discord, or to set up others! But I wanted to explain why I personally won't continue using it and hope others are feeling compelled by the above reasoning to switch too...

With all that said, I'm looking forward to find old & new people in this forum and learn what everyone's been up to (or want to build) with these projects... Come & say Hi! 👋

**Comment by acarabott** [comment]
Hey all, my name's Arthur and I've been using the umbrella libraries since 2018, primarily to prototype  music software products. In particular the interaction design and user interfaces. 

Most of that work is private (company), but my first umbrella project is public: [www.designsound.app](www.designsound.app) this was used as part of a talk to explain some basic synthesis concepts that impact the sound of an instrument.

I also have a talk at the Audio Developer Conference about my prototyping approach, and how I integrate it with C++ audio apps: [https://youtu.be/nRAQ2a7ghYA](https://youtu.be/nRAQ2a7ghYA)

I tend to use the hdom, hdom-canvas, atom, transducer, stream, math, and vector packages the most.

**Comment by JeffreyPalmer** [comment]
👋 I'm Jeff Palmer, and I use the umbrella libs to support my generative art practice, which you can see some examples of here: https://jpalmer.art. I love the depth of features that these libraries provide, and I love the anti-framework mindset that helps keep my project sizes manageable. 🙏 

Looking forward to meeting more folks!

**Comment by robertoranon** [comment]
Hi everybody, I'm Roberto Ranon, from Udine, Italy.
About me, briefly:
- 20+ years in the academia as [assistant professor/researcher](http://users.dimi.uniud.it/~roberto.ranon/) in the areas of HCI, 3D Graphics
- since 2017 (and full-time since 2021),[ independent consultant / developer on 3D Graphics, VR, AR and related technologies ](https://www.linkedin.com/in/robertoranon/)
- generative art is my main hobby, if you're curious find links at [linktr.ee/robertoranon](https://linktr.ee/robertoranon)

I have started using umbrella in the last activity in the past few months, very happy with what it enables me to create. I also plan to use it in my professional activities, which in the last couple of years have been 99% web-based (most of them cannot be made public here due to damn NDAs).

@postspectacular, I very much welcome your decision to move dev discussion away from discord

**Comment by Chickensoupwithrice** [comment]
Hi everyone, I'm Anish, still learning the dark arts of generative artistry, though I'm primarily interested in the Clojure version of Thi.ng rather than the typescript version (being a lisp fanatic and all) though I'm not skilled with visual pipelines yet to be able to just dive in, so I'm constantly on the lookout for tutorials / guides (my background being primarily DevOps) 

I run my [website](https://anish.lakhwara.com/) as a sort of catalog of everything I do, built entirely from the notes I take during my computer touching adventures. Asides from generative art, I run a bunch of services (though primarily Matrix) under https://sealight.xyz
I have Opinions™ about software, and laud the move away from Discord, though I'm not sure GH-Discussions is much of an improvement, though at the same time I understand it takes a lot of effort to run a community and people have different priorities. 

Super interested to connect with and learn from the community, I love meeting new people! I'm on mastodon, matrix and of course, email. :3

**Comment by Hantsouski** [comment]
Hi, I'm Yury. I found out about Karsten and his work a year ago. I admire all the hard work he's been putting into it. Wishing him all the best.

For the past year, I didn't have enough time to engage more with the tools and concepts to be able to help with something. But still, I have had the time to poke around the umbrella. Mostly I explored: rdom, rdom-canvas, rstream, transducers and geom. I learned the basics by looking at examples and played with it locally. If I didn't understand something, I would check interfaces and read comments in the source code, which is very helpful, but it also helps that I work with Angular, so working with streams and web api looks familiar to me. As well, I read Karsten's articles and workshops reports.

Recently, I started building a messenger app as a pet project just to learn how I can build more complex thing using these tools, but also to engage more with it and to find difficulties and opportunities to improve it. Hope to finish it and share it with Karsten :)

But most of all, I find his piece on [creating generative art](https://github.com/thi-ng/blog/blob/main/2022/20221204-personal-considerations.md) inspiring. It is my long term goal to start building something for myself to check if it can become my passion because I need something complex and meaningful to pursue.
